presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java

```diff
@@ -176,23 +176,36 @@ else if (frameInfo.getType() == RANGE) {
 
     private boolean emptyFrame(int rowPosition, int endPosition)
     {
-        if (frameInfo.getStartType() != frameInfo.getEndType()) {
+        FrameBound.Type startType = frameInfo.getStartType();
+        FrameBound.Type endType = frameInfo.getEndType();
+
+        int positions = endPosition - rowPosition;
+
+        if (startType == UNBOUNDED_PRECEDING && endType == PRECEDING) {
+            long end = getEndValue();
+            return end > rowPosition;
+        }
+
+        if (startType == FOLLOWING && endType == UNBOUNDED_FOLLOWING) {
+            long start = getStartValue();
+            return start > positions;
+        }
+
+        if (startType != endType) {
```

Comment:
The first two checks special-case the first one that returns false. The alternative is startType != endType && (a whole nasty mess to handle the special cases).